### PR TITLE
Fixd bug for pyenv plugin

### PR DESCRIPTION
**Pull request by Superbil**
Check pyenv-virtualenv can eval before use it

### CODE REVIEW

**Comment by arvindch** [comment]
Oops, I totally forgot that the pyenv virtualenv plugin is not installed by default.

However, there is no pyenv-virtualenv-init command, so this won't work...

Instead, this will:

```zsh
if [[ "$(pyenv commands | grep virtualenv-init -c)" == "1" ]]
then
    eval "$(pyenv virtualenv-init - zsh)"
fi
```

**Comment by Superbil** [comment]
@arvindch I found that command from `brew info pyenv-virtualenv`, so I use that package and test on it.

Look like check `pyenv commands` is a better way, I will fix my commit.

**Comment by arvindch** [comment]
Oh, interesting. It must be an OSX thing, since it's not on my Arch system. Anyway, yeah, this works on both, so cool.

File: plugins/pyenv/pyenv.plugin.zsh [hunk 1]
```diff
@@ -18,7 +18,10 @@ for pyenvdir in "${pyenvdirs[@]}" ; do
         export PYENV_ROOT=$pyenvdir
         export PATH=${pyenvdir}/bin:$PATH
         eval "$(pyenv init - zsh)"
-        eval "$(pyenv virtualenv-init - zsh)"
+
+        if [[ "$(pyenv commands | grep virtualenv-init -c)" == "1" ]] ; then
```
Comment (sachin21) [inline plugins/pyenv/pyenv.plugin.zsh]: I think this is better:
```sh
if [[ "$(pyenv commands | grep virtualenv-init -c  > /dev/null 2>&1)" == `false` ]]; then
*---* then *---*
```
If grep is succeeded, It'll see result, so I think that code is not good.

File: plugins/pyenv/pyenv.plugin.zsh [hunk 2]
```diff
@@ -18,7 +18,10 @@ for pyenvdir in "${pyenvdirs[@]}" ; do
         export PYENV_ROOT=$pyenvdir
         export PATH=${pyenvdir}/bin:$PATH
         eval "$(pyenv init - zsh)"
-        eval "$(pyenv virtualenv-init - zsh)"
+
+        if [[ "$(pyenv commands | grep virtualenv-init -c  > /dev/null 2>&1)" == `false` ]]; then
```
Comment (mcornella) [inline plugins/pyenv/pyenv.plugin.zsh]: A better alternative and common practice is using `grep`'s `-q` (quiet) argument. The final version should be:
```zsh
if pyenv commands | grep -q virtualenv-init; then
```

Comment (Superbil) [inline plugins/pyenv/pyenv.plugin.zsh]: Cool ! This is more clear.

Comment (sachin21) [inline plugins/pyenv/pyenv.plugin.zsh]: I have not have that option.
I think the all version to be considered :)

Comment (sachin21) [inline plugins/pyenv/pyenv.plugin.zsh]: I propose this:
```
if pyenv commands | grep virtualenv-init > /dev/null 2>&1; then
```

**Comment by mcornella** [comment]
FWIW, `grep -q` is [part of the POSIX spec](http://www.gnu.org/software/grep/manual/grep.html#index-g_t_002do-42), so it should be everywhere.

**Comment by sachin21** [comment]
@mcornella I'm sorry :cry: 
I'm using grep alias is alias='ag'
I replaced to the_sliver_searcher lol lol lol

**Comment by mcornella** [comment]
@sachin21 for that we should make sure we're using the `grep` command with the `command` builtin. So @Superbil please update to:

```zsh
if pyenv commands | command grep -q virtualenv-init; then
```

after that, I think everybody should be accounted for.

**Comment by sachin21** [comment]
@mcornella That is certainly true.
That's good :D

**Comment by Superbil** [comment]
@mcornella Ok, I update it.

**Comment by sachin21** [comment]
:+1:

**Comment by mcornella** [comment]
Good job! Thank you for your time :smile:

**Pull request merged**